tket2-py/src/circuit/convert.rs

```diff
@@ -21,71 +23,130 @@ pub struct T2Circuit {
 
 #[pymethods]
 impl T2Circuit {
+    /// Cast a tket1 circuit to a [`T2Circuit`].
     #[new]
-    fn from_circuit(circ: PyObject) -> PyResult<Self> {
+    pub fn from_circuit(circ: &PyAny) -> PyResult<Self> {
         Ok(Self {
-            hugr: with_hugr(circ, |hugr| hugr)?,
+            hugr: with_hugr(circ, |hugr, _| hugr)?,
         })
     }
 
-    fn finish(&self) -> PyResult<PyObject> {
-        SerialCircuit::encode(&self.hugr)?.to_tket1_with_gil()
+    /// Cast the [`T2Circuit`] to a tket1 circuit.
+    pub fn finish<'py>(&self, py: Python<'py>) -> PyResult<&'py PyAny> {
+        SerialCircuit::encode(&self.hugr)?.to_tket1(py)
     }
 
-    fn apply_match(&mut self, rw: PyCircuitRewrite) {
+    /// Apply a rewrite on the circuit.
+    pub fn apply_match(&mut self, rw: PyCircuitRewrite) {
         rw.rewrite.apply(&mut self.hugr).expect("Apply error.");
     }
+
+    /// Encode the circuit as a HUGR json string.
+    pub fn to_hugr_json(&self) -> PyResult<String> {
+        Ok(serde_json::to_string(&self.hugr).unwrap())
```

Comment:
a note that previously we have used messagepack (and the behaviour between json and messagepack is not always identical), but I'm ok avoiding the extra python dependency here for now - at least until the serialized format gets a bit more stable (hopefully soon)

Reply:
Added a TODO